### ISSUE TITLE
SC2093 about removing "exec" should trigger in loops

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1340]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2093
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
for file in rc.d/*
do
     exec "$file"
done
```

#### Here's what shellcheck currently says:

No issues

#### Here's what I wanted or expected to see:

SC2093: Remove "exec " if script should continue after this command.

ShellCheck does not emit this warning when the command is the last in its block, but it should still be triggering for loops.
